kotlin-analysis/src/main/kotlin/org/jetbrains/dokka/analysis/AnalysisEnvironment.kt

```diff
@@ -114,8 +114,12 @@ class AnalysisEnvironment(val messageCollector: MessageCollector, val analysisPl
             JavadocTagInfo.EP_NAME, JavadocTagInfo::class.java
         )
 
+        // TODO [beresnev] change to ApplicationManager#getExtensionArea after update to 213 platform
+        @Suppress("DEPRECATION")
+        val extensionArea = Extensions.getRootArea()
```

Comment:
The suggested replacement is `ComponentManager#getExtensionArea`, but the problem is that it needs to be an _instance_ of `ComponentManager` such as `ApplicationManager`, the method is not static.

I've investigated it for some time and got it to work correctly, but I don't understand why it works, so I'm afraid it'll cause bugs - decided not to change it here and now. Ideally, I'd like to have someone from the Intellij team to have a look at how we register extension points and services - I have a feeling we might be doing something wrong.

What concerns me is that we register some extension points in one extension area, and then we register/check other extension points and services in other extention areas, but for some reason it all works together. If you try to unify where extentions are registered, it introduces bugs and fails the tests.

Anyhow, we'll have to update to platform 213 soon (211 is not going to be updated anymore), I think then we'll have time to dig further.

Reply:
Sad, will remove the TODO before merging then

Reply:
But if you want, `(ApplicationManager.getApplication() as MockApplication).extensionArea ` ( the is the same `==Extensions.getRootArea()` ) can be used